lib/events-reader.js

```diff
@@ -74,6 +80,22 @@ module.exports = function(harvesterApp) {
       var doc;
       doc = that.stream.read();
 
+      // If it is a doc add to kafka
+      if(doc !== null && (!doc.ns.includes('checkpoint'))) {
+        producer.init().then(function(){
+          return producer.send({
+            topic: 'darksouls',
+            partition: 0,
+            message: {
+              value: JSON.stringify(doc)
+            }
+          });
+        })
+        .then(function (result) {
+          console.log('>>> producer sent message');
+        });
+      }
+
       var promises = that.processDocHandlers(doc);
```

Comment:
I'm a little confused by your language here. Yes processDocHandlers calls fns defined in onChange hooks. Each promise (returned from the fns from onChange hooks) should return after the onChange hook has successfully executed. I think you're asking if processDocHandlers does pretty much all the work to call the onChange clauses - and yes, it does, so you could move that to another file and separate consuming from kafka & processing events from streaming those events into kafka.